clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -386,13 +386,29 @@ class CIRGenBuilderTy : public CIRBaseBuilderTy {
     llvm_unreachable("NYI");
   }
 
-  mlir::cir::VectorType getExtendedElementVectorType(mlir::cir::VectorType vt,
-                                                     bool isSigned = false) {
+  mlir::cir::IntType getTruncatedIntTy(mlir::cir::IntType ty, bool isSigned) {
+    if (isInt16Ty(ty)) {
```

Comment:
Please write a switch on the width instead, and make the return statements to create the new integer directly, without using the helper functions. Too much indirection for something that should be fast, even though we could be reusing code here. While here, please also make sure to change `getExtendedIntTy` to be implemented in 
 similar fashin. Thanks!

Reply:
make sense, fixed